modules/dcache-nfs/src/main/java/org/dcache/chimera/nfsv41/door/proxy/NfsProxyIoFactory.java

```diff
@@ -78,29 +78,36 @@ public ProxyIoAdapter getOrCreateProxy(Inode inode, stateid4 stateid, CompoundCo
           boolean isWrite) throws IOException {
         try {
             return _proxyIO.get(stateid,
-                  () -> {
-                      final NFS4Client nfsClient;
-                      if (context.getMinorversion() == 1) {
-                          nfsClient = context.getSession().getClient();
-                      } else {
-                          nfsClient = context.getStateHandler().getClientIdByStateId(stateid);
-                      }
-
-                      final NFS4State state = nfsClient.state(stateid);
-                      final ProxyIoAdapter adapter = createIoAdapter(inode, stateid, context,
-                            isWrite);
-
-                      state.addDisposeListener(s -> {
-                          tryToClose(adapter);
-                          _proxyIO.invalidate(s.stateid());
-                      });
-
-                      return adapter;
-                  });
-        } catch (ExecutionException e) {
-            Throwable t = e.getCause();
-            _log.debug("failed to create IO adapter: {}", t.getMessage());
-            throw asNfsException(t, NfsIoException.class);
+                  (key) -> {
+                try {
+                    final NFS4Client nfsClient;
+                    if (context.getMinorversion() == 1) {
+                        nfsClient = context.getSession().getClient();
+                    } else {
+                        nfsClient = context.getStateHandler().getClientIdByStateId(stateid);
+                    }
+
+                    final NFS4State state = nfsClient.state(stateid);
```

Comment:
the sateid in this context is the `key`, isn't it?

Reply:
Yes it is, however I do not see any significant advantage (one can argue performance benefits for either or) of using the lambda parameter or the parameter from the higher scope. The idea is that you could use a getter or a return method in the statement and don't need to store it another variable

Advantage in the following case:
```java
cache.get(someinstance.getStringIDontWantToStoreInAVariable(), (string) -> "defaultValue" + string)
```

No real advantage in the following case (our case):
```java
public getFromCacheOrDefault(String key) {
    cache.get(key, (key_) -> "defaultValue" + key_) // I could just as easily use key and not key_ here.
}
```

Maybe I am mistaken, but I thought in this case it would be easier to just wait for JDK 21 where we have unnamed variables ("_"), hence I just called it key.